modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SClass.java

```diff
@@ -135,29 +136,30 @@ public void extractVariables(Set<String> variables) {
     }
 
     public void analyze(PainlessLookup painlessLookup) {
-        Map<String, LocalMethod> methods = new HashMap<>();
+        table = new FunctionTable();
 
         for (SFunction function : functions) {
             function.generateSignature(painlessLookup);
 
-            String key = Locals.buildLocalMethodKey(function.name, function.parameters.size());
+            String key = FunctionTable.buildLocalFunctionKey(function.name, function.parameters.size());
 
-            if (methods.put(key,
-                    new LocalMethod(function.name, function.returnType, function.typeParameters, function.methodType)) != null) {
-                throw createError(new IllegalArgumentException("Duplicate functions with name [" + function.name + "]."));
+            if (table.getFunction(key) != null) {
+                throw createError(new IllegalArgumentException("Illegal duplicate functions [" + key + "]."));
```

Comment:
nit: can this message be something like "X is already defined"?

Reply:
Fixed.